sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -477,22 +476,18 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
   }
 
   test("withColumns: case sensitive") {
-    withSQLConf(SQLConf.CASE_SENSITIVE.key -> "true") {
-      val df = testData.toDF().withColumns(Seq("newCol1", "newCOL1"),
+    spark.sessionState.conf.setConf(SQLConf.CASE_SENSITIVE, true)
```

Comment:
I agree with the idea of this PR.

In the implementation, can we use the current APIs (e.g 'withSQLConf') as possible?
It is easy to see the scope of the specified option and is less to change # of lines.

Reply:
I agree with the idea of this PR in general and this comment.

Can we consider this new copy just as a safe guard? I think it's fine to clean up within the test.

Reply:
In most of the cases, one test only set the config once, so scoping is not a big issue here. My goal is to let us completely forget about the cleanup and write tests easily.

Reply:
BTW we can always create a scope if we want to
```
setConf...
{
  ...
}
setConf...
{
  ...
}
```

Reply:
scoping is fine but I just wonder if we can have less changes when we replace `SharedSparkSession` to `SharedSparkSessionWithFreshCopy`.

It would be nice if we only need to replace `SharedSparkSession` to `SharedSparkSessionWithFreshCopy` in class definition to use this trait.

Reply:
> I just wonder if we can have less changes when we replace SharedSparkSession to SharedSparkSessionWithFreshCopy.

Yes we can. But I want to demonstrate how `SharedSparkSessionWithFreshCopy` simplifies the tests, and that's why I made so many changes to `DataFrameSuite`.

Later when I add new tests to an existing test suite, I will use `SharedSparkSessionWithFreshCopy` without touching the existing tests, but stop using `withSQLConf`, `withTempView` in the newly added test cases.